src/utils/dif_upload.rs

```diff
@@ -1101,16 +1105,14 @@ fn upload_difs_chunked(
         ChunkedDifMatch::from(m, chunk_options.chunk_size)
     })?;
 
-    // Upload until all chunks are present on the server
-    let mut initially_missing = None;
-    while let Some(missing_info) = try_assemble_difs(&chunked, options)? {
-        upload_missing_chunks(&missing_info, chunk_options)?;
-        initially_missing.get_or_insert(missing_info);
-    }
+    // Upload missing chunks to the server and remember incomplete difs
+    let missing_info = try_assemble_difs(&chunked, options)?;
+    upload_missing_chunks(&missing_info, chunk_options)?;
```

Comment:
We previously looped here to be defensive, but there is no real reason. Since we're uploading all chunks, there won't be any missing afterwards. Upload errors will exit immediately in both cases.